nightfall-client/src/services/state-sync.mjs

```diff
@@ -44,14 +53,73 @@ const syncState = async (fromBlock = 'earliest', toBlock = 'latest', eventFilter
   }
 };
 
+const downloadFile = async (fileUrl, outputLocationPath) => {
+  const writer = fs.createWriteStream(outputLocationPath);
+  return axios({
+    method: 'get',
+    url: fileUrl,
+    responseType: 'stream',
+  }).then(response => {
+    response.data.pipe(writer);
+    return finished(writer); // this is a Promise
+  });
+};
+
 const genGetCommitments = async (query = {}, proj = {}) => {
   const connection = await mongo.connection(MONGO_URL);
   const db = connection.db(COMMITMENTS_DB);
   return db.collection(COMMITMENTS_COLLECTION).find(query, proj).toArray();
 };
 
+const checkContractsABI = async () => {
+  let env = '';
+  switch (ETH_NETWORK) {
+    case 'goerli':
+      env = 'testnet';
+      break;
+    case 'mainnet':
+      env = 'production';
+      break;
+    default:
+      env = '';
+  }
+
+  if (env) {
+    const baseUrl = CONTRACT_FILES_URL
+      ? `${CONTRACT_FILES_URL}`
+      : `${DEFAULT_CONTRACT_FILES_URL}/${env}`;
+    const url = `${baseUrl}/build/hash.txt`;
+
+    const res = await axios.get(url); // get all json abi contracts
+    const files = res.data.split('\n');
+
+    if (!fs.existsSync(`${config.CONTRACT_ARTIFACTS}`)) {
+      fs.mkdirSync(`${config.CONTRACT_ARTIFACTS}`);
+    }
+
+    logger.info(`Downloading contracts from ${url}...`);
+
+    await Promise.all(
+      files.map(async f => {
+        if (f) {
+          try {
+            await downloadFile(
+              `${baseUrl}/build/contracts/${f.split('  ')[1]}`,
+              `${config.CONTRACT_ARTIFACTS}/${f.split('  ')[1]}`,
+            );
+          } catch (e) {
+            console.error(`ERROR downloading ${f.split('  ')[1]}`);
```

Comment:
Here we just log that an error happened. Is the download of a contract required for the app to work as expected? if so, I think we should rethrow the error.